src/flac/main.c

```diff
@@ -1278,98 +1278,109 @@ void show_help(void)
+	printf("      --ignore-chunk-sizes           Ignore data chunk sizes in WAVE/AIFF files\n");
+	printf("      --replay-gain                  Calculate ReplayGain & store in FLAC tags\n");
+	printf("      --cuesheet=FILENAME            Import cuesheet & store in CUESHEET block\n");
+	printf("      --picture=SPECIFICATION        Import picture & store in PICTURE block\n");
+	printf("  -T, --tag=FIELD=VALUE              Add a FLAC tag; may appear multiple times\n");
+	printf("      --tag-from-file=FIELD=FILENAME     Like --tag but gets value from file\n");
+	printf("  -S, --seekpoint={#|X|#x|#s}        Add seek point(s)\n");
+	printf("  -P, --padding=#                    Write a PADDING block of length # bytes\n");
+	printf("\n");
+	printf("Format options (encoding defaults to FLAC not OGG; decoding defaults to WAVE, \n");
+	printf("             chunks found by --keep-foreign-metadata-if-present will override):\n");
+#if FLAC__HAS_OGG
+	printf("      --ogg                          Use Ogg transport layer, output .oga\n");
+	printf("      --serial-number                Ogg serial number to assign (encoding)\n);
+	printf("                                     or to select for decoding\n);
+#endif
 	printf("      --force-aiff-format            Decode to AIFF format\n");
 	printf("      --force-rf64-format            Decode to RF64 format\n");
 	printf("      --force-wave64-format          Decode to Wave64 format\n");
 	printf("      --force-legacy-wave-format     Decode to legacy wave format\n");
 	printf("      --force-extensible-wave-format Decode to extensible wave format\n");
 	printf("      --force-aiff-c-none-format     Decode to AIFF-C NONE format\n");
 	printf("      --force-aiff-c-sowt-format     Decode to AIFF-C sowt format\n");
-	printf("raw format options:\n");
-	printf("      --endian={big|little}    Set byte order for samples\n");
-	printf("      --channels=#             Number of channels\n");
-	printf("      --bps=#                  Number of bits per sample\n");
-	printf("      --sample-rate=#          Sample rate in Hz\n");
-	printf("      --sign={signed|unsigned} Sign of samples\n");
-	printf("      --input-size=#           Size of the raw input in bytes\n");
-	printf("negative options:\n");
+	printf("      --force-raw-format             Treat input or output as raw samples\n");
+	printf("raw format options mandatory for encoding; --sign and --endian also for decoding\n");
+	printf("      --sign={signed|unsigned}       Sign of samples\n");
+	printf("      --endian={big|little}          Byte order for samples\n");
+	printf("      --channels=#                   Number of channels in raw input\n");
+	printf("      --bps=#                        Number of bits per sample in raw input\n");
+	printf("      --sample-rate=#                Sample rate in Hz in raw input\n");
+	printf("      --input-size=#                 Size of the raw input in bytes\n");
+	printf("\n");
+	printf("Negative options (rightmost applied takes precedence):\n");
```

Comment:
I'm not sure this is best placed here, because it applies to all options, not just negative ones. AFAIK `flac -8l32` is different from  `flac -l32 -8`. Maybe add a note at the top, before general options, something like: `the order in which options are applied is important, options applied last take precedence over options applied first.

Reply:
This is maybe a new issue, combining -d, -t and -a don't work that way. And now -t makes checks -d don't.

Reply:
Hmm, interesting. I usually don't like changing behaviour because it breaks scripts, but this seems wrong. If I use `-t` and `-a` together I get a warning that that combination is not possible. Maybe I should add the same for any combination of `-d`, `-t` and `-a`?

Reply:
I think this is turning into a new issue especially with the latter paragraph here, but to pursue the train of thought:

If anyone has used -t and -d together - say, mistakenly thinking that -dt is _the_ test-decode option - then erring out on that might break scripts too. As for other combinations I see that -da and -ad both work as -a. And -t -c (which isn't too sensible?!) works as -t. One may change it to "last one decides", but let me suggest two or three other ways out: 

- One is to `-td` or `-ta` work as -d or -a would, but invoke the extra tests of `-t`. Change is biggest for -td which as of now doesn't output any file. It can be "rationalized" by saying that "-t switches to decode with all tests, but switches off output; -d / -a will switch on output but not switch off the tests". With that logic, `-dt` should work as currently. Then `-at` will either have to change (but currently it errs out, so how many use it?) or stick with some inconsistency. 
- The other is to let -d and -a actually do those extra tests that -t does, but only report warning. Unexpected to those who invoke -w?
- Probably not a good idea, but still thinking aloud: had you not already made -t more zealous (reverting that is maybe not a good idea at this stage), you could have considered the -V switch. Provided that I understand correctly that -V does nothing except when encoding? There is no natural analog to decoding-the-encode, but had one started from scratch, -tV or -dV or -aV could have been the extra zealous one that goes to greater lengths to catch all errors. But again, with the recent changes to -t, it might just be dumb to do that?


Since this brings up how much checks -d and -a should do, it is maybe natural to ask for the demand for a "fast test" option that doesn't decode. As you have pointed out elsewhere, FLAC is maybe the codec least in need for this since it decodes so fast, but still even OptimFROG has a verification that is faster than FLAC's. I was initially thinking that one could implement some `-t --no-md5-sum` for non-decoding verification (because what do you need the audio for if not to MD5 it?), but if I am right, -t checks more about the audio than the MD5? (Like, decoded values don't exceed -2^(N-1) ... 2^(N-1)-1?) 
Anyway I think that if you consider to change what is tested, then a fast-verification option could also be part of the consideration.

Reply:
> If anyone has used -t and -d together - say, mistakenly thinking that -dt is _the_ test-decode option - then erring out on that might break scripts too.

Yes, but I think breaking scripts isn't a bad thing here.